media/js/base/mozilla-fxa-iframe.js

```diff
@@ -72,6 +72,7 @@ Mozilla.FxaIframe = (function() {
     };
 
     var _sendGAEvent = function(type, extra) {
+        var dataLayer = window.dataLayer || [];
```

Comment:
OK, it looks like the test that fails due to `TypeError: window.dataLayer is undefined` was relying on where the declaration is being removed from `mozilla-client.js` above  :/

I think this is fine to leave in for now, although I'd like to find a more robust way to mock `window.dataLayer` for all tests in the future.